source/developer-guides/configuration.rst

```diff
@@ -243,3 +246,179 @@ External configuration data for board (platform 1) is loaded::
   Load Graphics Cfg Data
   ...
 
+.. _gpio-configuration:
+
+GPIO Configuration
+^^^^^^^^^^^^^^^^^^
+
+GPIO / CFIO
+"""""""""""
+
+Modern chipsets pack multiple functions and are often constrained by the package 
+size and limited number of pins. These Configurable IO (CFIO) pins need to be configured 
+at boot time to connect them to selected functionality based on platform designs.
+
+The selected functionality could be a General Purpose IO (GPIO) or some other 
+alternative function (SPI, I2C, etc.)
+
+General Purpose IO (GPIO) is a digital or analog signal pin and can be configured 

```

Comment:
@atharvalele it's very rare GPIO can work as analog signal, could you confirm it? at minimal it's not an option on Intel platform.

Reply:
@tsaikevin - just checked and you're right. I removed the "analog" part from the doc. Thanks!